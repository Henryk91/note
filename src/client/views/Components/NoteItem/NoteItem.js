import React, { Component } from 'react'

export default class NoteItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
      item: this.props.item,
      editingItem: false,
    }
    this.deleteItem = this.deleteItem.bind(this)
  }

  deleteItem = (e) => {
    e.preventDefault();
    if (confirm("Are you sure you want to permanently delete this?")) {
      this.setState({ item: null })
      this.props.set({ oldItem: this.state.item, index: this.props.index , type: this.props.type , delete: true}) 
    }
  }

  editItem = () => {
    this.setState({ editingItem: true })
  }

  submitChange = (e) => {
    e.preventDefault();
    let update = e.target.item.value
    this.props.set({item: update, oldItem: this.state.item, index: this.props.index , type: this.props.type , delete: false}) 
    this.setState({ editingItem: false , item: update})
  }

  editItemBox(item) {

    return (
      <form onSubmit={this.submitChange} className="noteItemBox" id="noteItemEditBox">
        <input name="item" type="text" defaultValue={item}></input>
        <button type="submit">Submit</button>
        <button onClick={() => this.setState({ editingItem: false })}>Cancel</button>
      </form>
    )
  }

  displayItemBox(item){
    return (
      <div className="noteItemBox">
        <p className="noteItem">{item}</p>
        <button onClick={() => this.setState({ editingItem: true })}>Edit</button>
        <button  onClick={this.deleteItem}>Del</button>
        <br />
      </div>
    )
  }

  render() {
    const item = this.state.item
    let editing = this.state.editingItem
    return (
      <div key={item}>
        {item ? <div className="noteTagBox">{editing ? this.editItemBox(item) : this.displayItemBox(item)}</div> : null}
      </div>
    )
  }
}
