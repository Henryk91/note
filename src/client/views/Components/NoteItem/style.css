.noteItemBox{
    margin:0;
    width:100%;
}
.noteItemBox > button{
    float: right;
    margin: 0;
    margin-left: 5px;
}
.noteItem{
    margin: 4px;
    padding: 0px;
}
#noteItemEditBox > button{
    margin-top: 5px; 
    float: none
}
#noteItemEditBox{
    margin-bottom: 10px; 
    height: 60px;
    text-align: center;
}
