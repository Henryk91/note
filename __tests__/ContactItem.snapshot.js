
import NoteItem from '../src/client/views/Components/NoteItem/NoteItem';
import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';


test('NoteItem component renders all details', () => {

    const rendered = renderer.create(
        <Router>
            <NoteItem item={"0808080"} type={"Number"} index={0} />
        </Router>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
});