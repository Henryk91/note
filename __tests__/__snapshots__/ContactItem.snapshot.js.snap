// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteItem component renders all details 1`] = `
<div>
  <div>
    <div
      className="noteItemBox"
    >
      <p
        className="noteItem"
      >
        0808080
      </p>
      <button
        onClick={[Function]}
      >
        Edit
      </button>
      <button
        onClick={[Function]}
      >
        Del
      </button>
      <br />
    </div>
  </div>
</div>
`;
